File: frontend/bookstore-front/src/app/pages/livro/models/livro.ts
import { Assunto } from "../../assunto/models/assunto";
import { Autor } from "../../autor/models/autor";

export interface Livro {
    id: number;
    titulo: string;
    editora: string;
    edicao: number;
    anoPublicacao: string;
    valor: number;
    assuntos: Assunto[];
    autores: Autor[];
}

File: backend/bookstore-back/src/BookStore.Business/Models/Livro.cs
namespace BookStore.Business.Models;

public class Livro : Entity
{
    public string Titulo { get; set; }
    public string Editora { get; set; }
    public int Edicao { get; set; }
    public string AnoPublicacao { get; set; }
    public decimal Valor { get; set; }
    public ICollection<Assunto> Assuntos { get; set; } = [];
    public ICollection<Autor> Autores { get; set; } = [];
}
